core-web/libs/portlets/edit-ema/portlet/src/lib/utils/index.ts

```diff
@@ -222,3 +230,126 @@ export function createPageApiUrlWithQueryParams(
 export function getIsDefaultVariant(variant?: string): boolean {
     return !variant || variant === DEFAULT_VARIANT_ID;
 }
+
+/**
+ * Check if the param is a forward or page
+ *
+ * @export
+ * @param {VanityUrl} vanityUrl
+ * @return {*}
+ */
+export function isForwardOrPage(vanityUrl?: VanityUrl): boolean {
+    return !vanityUrl || (!vanityUrl.permanentRedirect && !vanityUrl.temporaryRedirect);
+}
+
+/**
+ * Create the url to add a page to favorites
+ *
+ * @private
+ * @param {{
+ *         languageId: number;
+ *         pageURI: string;
+ *         deviceInode?: string;
+ *         siteId?: string;
+ *     }} params
+ * @return {*}  {string}
+ * @memberof EditEmaStore
+ */
+export function createFavoritePagesURL(params: {
+    languageId: number;
+    pageURI: string;
+    siteId: string;
+}): string {
+    const { languageId, pageURI, siteId } = params;
+
+    return (
+        `/${pageURI}?` +
+        (siteId ? `host_id=${siteId}` : '') +
+        `&language_id=${languageId}`
+    ).replace(/\/\//g, '/');
+}
+
+export function createPureURL(params: DotPageApiParams): string {
+    // If we are going to delete properties from the params, we need to make a copy of it
+    const paramsCopy = { ...params };
+
+    const clientHost = paramsCopy.clientHost ?? window.location.origin;
+    const url = paramsCopy.url;
+
+    // Clean the params that are not needed for the page
+    delete paramsCopy.clientHost;
+    delete paramsCopy.url;
+    delete paramsCopy.mode;
+
+    const searchParams = new URLSearchParams(paramsCopy as unknown as Record<string, string>);
+
+    return `${clientHost}/${url}?${searchParams.toString()}`;
+}
+
+export function computeCanEditPage(
+    pageAPIResponse: DotPageApiResponse,
+    currentUser: CurrentUser,
+    experiment?: DotExperiment
+): boolean {
+    const pageCanBeEdited = pageAPIResponse.page.canEdit;
+    const isLocked =
+        pageAPIResponse.page.locked && pageAPIResponse.page.lockedBy !== currentUser?.userId;
+    const editingBlockedByExperiment = [
+        DotExperimentStatus.RUNNING,
+        DotExperimentStatus.SCHEDULED
+    ].includes(experiment?.status);
+
+    return pageCanBeEdited && !isLocked && !editingBlockedByExperiment;
+}
+
+/**
+ * Map the containers to a DotContainerMap
+ *
+ * @private
+ * @param {DotPageContainerStructure} containers
+ * @return {*}  {DotContainerMap}
+ */
+export function mapContainers(containers: DotPageContainerStructure): DotContainerMap {
+    return Object.keys(containers).reduce((acc, id) => {
+        acc[id] = containers[id].container;
+
+        return acc;
+    }, {});
+}
+
+/**
+ * Get the containers data
+ *
+ * @private
+ * @param {ContainerData} containers
+ * @memberof EditEmaStore
+ */
+export const getPageContainers = (containers: DotPageContainerStructure) => {
```

Comment:
Can we find a better naming here? is like we're passing containers and getting containers.